utils/src/compress/zstd_standard.rs

```diff
@@ -0,0 +1,63 @@
+use std::io::Result;
+
+use libc::{c_uint, c_void};
+use std::ffi::CStr;
+use zstd_sys::{
+    ZSTD_compress, ZSTD_compressBound, ZSTD_decompress, ZSTD_defaultCLevel, ZSTD_getErrorName,
+    ZSTD_isError,
+};
+
+pub(super) fn zstd_compress(src: &[u8]) -> Result<Vec<u8>> {
+    let compress_bound = unsafe { ZSTD_compressBound(src.len()) };
+
+    if src.len() > (i32::max_value() as usize) || compress_bound == 0 {
+        return Err(einval!("compression input data is too big"));
+    }
+
+    let mut dst_buf = Vec::with_capacity(compress_bound as usize);
+    let cmp_size = unsafe {
+        ZSTD_compress(
+            dst_buf.as_mut_ptr() as *mut c_void,
+            compress_bound,
+            src.as_ptr() as *const c_void,
+            src.len(),
+            ZSTD_defaultCLevel(),
```

Comment:
For chunks `<=1MB`, is the default compression level optimal for performance? @liubogithub @hsiangkao wdyt?

Reply:
TBH, I have no idea, a few performance tests is needed to prove here.
IMO it can be also updated later once we get some reasonable numbers here.

Reply:
> For chunks `<=1MB`, is the default compression level optimal for performance? @liubogithub @hsiangkao wdyt?

My own suggestion is to revise chunk compression approach later by introducing another compressed segment concept, so each compressed segment can roughly have 1MB size.
I don't quite like the current compression approach, it's somewhat ineffective.

Reply:
but I think currently we can live with that, since currently compression algorithm is selected on the fs basis. so we have no other choice, so the compression level is not a on-disk chance. We can live with it for now, and benchmark even further later.

Reply:
Ok, let's keep going on.